src/effect_on_condition.cpp

```diff
@@ -91,20 +95,32 @@ static time_duration next_recurrence( const effect_on_condition_id &eoc )
 void effect_on_conditions::load_new_character( Character &you )
 {
     bool is_avatar = you.is_avatar();
+    for( const effect_on_condition_id &eoc_id : get_scenario()->eoc() ) {
+        effect_on_condition eoc = eoc_id.obj();
+        if( eoc.scenario_specific ) {
+            queued_eoc new_eoc = queued_eoc{ eoc.id, true, calendar::turn + next_recurrence( eoc.id ) };
+            you.queued_effect_on_conditions.push( new_eoc );
+        }
+    }
```

Comment:
Right now this will run for NPC's as well.  I'm pretty sure you want to put an if (is_avatar) here.